### DISCUSSION TITLE
should `type[bool]` and `type[int]` be considered overlapping?

### DISCUSSION THREAD

**Discussion by Dr-Irv** [microsoft/pyright#4679] [General]
See https://github.com/python/mypy/issues/14772

Would like to hear the opinion of @erictraut here, as pyright does not flag this as an error (which is correct IMHO)

**Comment by erictraut** [comment]
These are overlapping overloads, and pyright reports them as such if you enable `reportOverlappingOverload`. Mypy is doing the right thing.
